### PR TITLE
allow "schedule" option in proposal creator

### PR DESCRIPTION
**Pull request by DZGoldman**

### CODE REVIEW

File: src-ts/proposalCreator.ts [hunk 1]
```diff
@@ -225,15 +226,28 @@ export class RoundTripProposalCreator {
...
-
-    const l1TimelockScheduleCallData = l1Timelock.interface.encodeFunctionData("scheduleBatch", [
-      l1Targets,
-      l1Values,
-      l1CallDatas,
-      constants.HashZero,
-      descriptionHash,
-      minDelay,
-    ]);
+    let l1TimelockScheduleCallData: string;
```
Comment (fredlacs) [inline src-ts/proposalCreator.ts:229]: could use an IIFE and make this a const :)

**Review by fredlacs — APPROVED** [review]
LGTM

**Review by yahgwai — APPROVED** [review]

File: src-ts/proposalCreator.ts [hunk 2]
```diff
@@ -153,7 +153,8 @@ export class RoundTripProposalCreator {
     upgradeAddrs: string[],
     upgradeValues: BigNumber[],
     upgradeDatas: string[],
-    proposalDescription: string
+    proposalDescription: string,
+    useSchedule = false // defaults to scheduleBatch in L1 timelock. If true, will use schedule. Can only be used if only one action is included in proposal
```
[the remark this answers is not in the document]
Comment (DZGoldman) [reply src-ts/proposalCreator.ts:157]: we would lose "backwards compatibility" with re-generating calldata for the older proposals tho

File: src-ts/proposalCreator.ts (hunk 1 above)
Comment (DZGoldman) [reply src-ts/proposalCreator.ts:229]: i thought you were anti-IIFE, maybe i'm misremembering tho, been awhile

**Pull request merged**